lib/reckon/app.rb

```diff
@@ -32,12 +38,25 @@ def already_seen?(row)
       seen[row[:pretty_date]] && seen[row[:pretty_date]][row[:pretty_money]]
     end
 
+    def extract_account_tokens(subtree, account = nil)
+      if subtree.is_a?(Array)
+        { account => subtree }
+      else
+        at = subtree.map { |k, v| extract_account_tokens(v, [account, k].compact.join(':')) }
+        at.inject({}) { |k, v| k = k.merge(v)}
+      end
+    end
+
     def learn!
-      if options[:existing_ledger_file]
-        fail "#{options[:existing_ledger_file]} doesn't exist!" unless File.exists?(options[:existing_ledger_file])
-        ledger_data = File.read(options[:existing_ledger_file])
-        learn_from(ledger_data)
+      if options[:account_tokens_file]
+        extract_account_tokens(YAML.load_file(options[:account_tokens_file])).each do |account, tokens|
```

Comment:
Might be nice to raise an error if `options[:account_tokens_file]` doesn't exist on disk?